### DISCUSSION TITLE
Phone Freezing with LSPosed 1.8.1+

### DISCUSSION THREAD

**Discussion by gamerkonks** [LSPosed/LSPosed#1997] [General]
I have a Google Pixel 3 XL on Android 12 with LSPosed 1.8.0
I have tried updating to 1.8.1, 1.8.2, 1.8.3 but they cause my phone to freeze and become completely unresponsive, so I have downgraded to 1.8.0, which is the latest version which doesn't cause this issue.
To be able to use my phone again I need to hard reboot.
The behaviour appears similar to a known issue when installing the Magisk Systemless Hosts module on a Google Pixel 3 / 3 XL.

**Comment by yujincheng08** [comment]
Thanks for your report. But unfortunately, it won't be fixed :).

**Reply by gamerkonks** [reply]
Okay, no worries.
Is this a known issue? What's causing it?

**Reply by yujincheng08** [reply]
we dont know because only you have such an issue and no logs provided.

**Reply by gamerkonks** [reply]
I have installed LSPosed-v1.8.3-6599-zygisk-debug and Log Catcher-v22.5(27) (2).
[boot-2022-08-24-11-55-54.log](https://github.com/LSPosed/LSPosed/files/9411795/boot-2022-08-24-11-55-54.log)
[magisk.log](https://github.com/LSPosed/LSPosed/files/9411796/magisk.log)

After how long does the device become unresponsive?
Within a minute or so.
Usually I'll have to open magisk as soon as possible and disable LSPosed then reboot to be able to use my phone again.
Sometimes I can open apps / close apps, navigate the UI for a little while (minutes), but the apps will only show the splash screen, not actually open. Eventually the UI will stop responding and the screen will be frozen. 

Can you connect with adb before become unresponsive?
Yes.

Please let me know if there is anymore information I can provide to help figure out what the issue is.
I've started having another issue recently (the only thing that's changed is that I've installed Shamiko), where LSPosed won't be installed anymore. To fix I need to install 1.8.3, then downgrade to 1.8.0 again.

**Reply by gamerkonks** [reply]
Some additional logs.
Sometimes the phone freezes before I can save the magisk logs.

[bootlog-2022-08-24-11-54-20.tar.gz](https://github.com/LSPosed/LSPosed/files/9411914/bootlog-2022-08-24-11-54-20.tar.gz)
[bootlog-2022-08-24-11-55-54.tar.gz](https://github.com/LSPosed/LSPosed/files/9411915/bootlog-2022-08-24-11-55-54.tar.gz)
[bootlog-2022-08-24-12-19-11.tar.gz](https://github.com/LSPosed/LSPosed/files/9411916/bootlog-2022-08-24-12-19-11.tar.gz)
[bootlog-2022-08-24-12-20-55.tar.gz](https://github.com/LSPosed/LSPosed/files/9411918/bootlog-2022-08-24-12-20-55.tar.gz)
[bootlog-2022-08-24-12-24-09.tar.gz](https://github.com/LSPosed/LSPosed/files/9411919/bootlog-2022-08-24-12-24-09.tar.gz)
[bootlog-2022-08-24-12-25-34.tar.gz](https://github.com/LSPosed/LSPosed/files/9411920/bootlog-2022-08-24-12-25-34.tar.gz)
[magisk_log_2022-08-24T11.56.06.log](https://github.com/LSPosed/LSPosed/files/9411921/magisk_log_2022-08-24T11.56.06.log)
[magisk_log_2022-08-24T12.21.37.log](https://github.com/LSPosed/LSPosed/files/9411922/magisk_log_2022-08-24T12.21.37.log)
[magisk_log_2022-08-24T12.25.43.log](https://github.com/LSPosed/LSPosed/files/9411923/magisk_log_2022-08-24T12.25.43.log)

**Reply by mywalkb** [reply]
@gamerkonks in your logs I see this entry:

`type=1400 audit(0.0:31828): avc: denied { read } for name="xposed" dev="dm-8" ino=33604 scontext=u:r:system_server:s0 tcontext=u:object_r:system_data_root_file:s0 tclass=file permissive=0`

repeated many times. I don't know because there is "xposed" I expect lspd.
Can you try to disable this module: ru.aeurutws.yhggslpbh?
Do you know it? Is it trusted?

It generate also this exception:

```
08-24 11:55:38.933  1906  1906 W System.err: java.io.FileNotFoundException: /data/local/tmp/xposed: open failed: EACCES (Permission denied)
08-24 11:55:38.934  1906  1906 W System.err:    at libcore.io.IoBridge.open(IoBridge.java:574)
08-24 11:55:38.934  1906  1906 W System.err:    at java.io.RandomAccessFile.<init>(RandomAccessFile.java:289)
08-24 11:55:38.934  1906  1906 W System.err:    at com.ʻ.ʿ.ˆ(Utils.java:7928)
08-24 11:55:38.934  1906  1906 W System.err:    at com.ʻ.ʿ.ﾞﾞ(Utils.java:11088)
08-24 11:55:38.934  1906  1906 W System.err:    at com.xposed.XSupport.ʻ(XSupport.java:1232)
08-24 11:55:38.934  1906  1906 W System.err:    at com.xposed.XSupport$13.beforeHookedMethod(XSupport.java:604)
08-24 11:55:38.934  1906  1906 W System.err:    at de.robv.android.xposed.XposedBridge$AdditionalHookInfo.callback(XposedBridge.java:451)
08-24 11:55:38.934  1906  1906 W System.err:    at LSPHooker_.assertPackageIsValid(Unknown Source:25)
08-24 11:55:38.934  1906  1906 W System.err:    at com.android.server.pm.PackageManagerService.scanPackageNewLI(PackageManagerService.java:13896)
08-24 11:55:38.934  1906  1906 W System.err:    at com.android.server.pm.PackageManagerService.addForInitLI(PackageManagerService.java:12248)
08-24 11:55:38.934  1906  1906 W System.err:    at com.android.server.pm.PackageManagerService.scanDirLI(PackageManagerService.java:11842)
08-24 11:55:38.934  1906  1906 W System.err:    at com.android.server.pm.PackageManagerService.scanDirTracedLI(PackageManagerService.java:11793)
08-24 11:55:38.934  1906  1906 W System.err:    at com.android.server.pm.PackageManagerService.<init>(PackageManagerService.java:7312)
08-24 11:55:38.934  1906  1906 W System.err:    at java.lang.reflect.Method.invoke(Native Method)
08-24 11:55:38.934  1906  1906 W System.err:    at org.lsposed.lspd.nativebridge.HookBridge.invokeOriginalMethod(Native Method)
08-24 11:55:38.934  1906  1906 W System.err:    at de.robv.android.xposed.XposedBridge$AdditionalHookInfo.callback(XposedBridge.java:471)
08-24 11:55:38.934  1906  1906 W System.err:    at LSPHooker_.constructor(Unknown Source:54)
08-24 11:55:38.934  1906  1906 W System.err:    at com.android.server.pm.PackageManagerService.main(PackageManagerService.java:6768)
08-24 11:55:38.934  1906  1906 W System.err:    at com.android.server.SystemServer.startBootstrapServices(SystemServer.java:1163)
08-24 11:55:38.934  1906  1906 W System.err:    at java.lang.reflect.Method.invoke(Native Method)
08-24 11:55:38.934  1906  1906 W System.err:    at org.lsposed.lspd.nativebridge.HookBridge.invokeOriginalMethod(Native Method)
08-24 11:55:38.934  1906  1906 W System.err:    at de.robv.android.xposed.XposedBridge$AdditionalHookInfo.callback(XposedBridge.java:471)
08-24 11:55:38.934  1906  1906 W System.err:    at LSPHooker_.startBootstrapServices(Unknown Source:11)
08-24 11:55:38.934  1906  1906 W System.err:    at com.android.server.SystemServer.run(SystemServer.java:877)
08-24 11:55:38.934  1906  1906 W System.err:    at com.android.server.SystemServer.main(SystemServer.java:610)
08-24 11:55:38.934  1906  1906 W System.err:    at java.lang.reflect.Method.invoke(Native Method)
08-24 11:55:38.934  1906  1906 W System.err:    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
08-24 11:55:38.934  1906  1906 W System.err:    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:981)
08-24 11:55:38.934  1906  1906 W System.err: Caused by: android.system.ErrnoException: open failed: EACCES (Permission denied)
08-24 11:55:38.934  1906  1906 W System.err:    at libcore.io.Linux.open(Native Method)
08-24 11:55:38.934  1906  1906 W System.err:    at libcore.io.ForwardingOs.open(ForwardingOs.java:563)
08-24 11:55:38.934  1906  1906 W System.err:    at libcore.io.BlockGuardOs.open(BlockGuardOs.java:274)
08-24 11:55:38.934  1906  1906 W System.err:    at libcore.io.IoBridge.open(IoBridge.java:560)
08-24 11:55:38.934  1906  1906 W System.err:    ... 27 more

```
or this exception:

```
08-24 11:55:48.047  3333  3571 D StrictMode: StrictMode policy violation: android.os.strictmode.DiskReadViolation
08-24 11:55:48.047  3333  3571 D StrictMode:    at android.os.StrictMode$AndroidBlockGuardPolicy.onReadFromDisk(StrictMode.java:1659)
08-24 11:55:48.047  3333  3571 D StrictMode:    at libcore.io.BlockGuardOs.access(BlockGuardOs.java:74)
08-24 11:55:48.047  3333  3571 D StrictMode:    at java.io.UnixFileSystem.checkAccess(UnixFileSystem.java:281)
08-24 11:55:48.047  3333  3571 D StrictMode:    at java.io.File.canRead(File.java:762)
08-24 11:55:48.047  3333  3571 D StrictMode:    at com.xposed.XSupport.ʻ(XSupport.java:1222)
08-24 11:55:48.047  3333  3571 D StrictMode:    at com.xposed.XSupport$15.beforeHookedMethod(XSupport.java:644)
08-24 11:55:48.047  3333  3571 D StrictMode:    at de.robv.android.xposed.XposedBridge$AdditionalHookInfo.callback(XposedBridge.java:451)
08-24 11:55:48.047  3333  3571 D StrictMode:    at LSPHooker_.getApplicationInfo(Unknown Source:25)
08-24 11:55:48.047  3333  3571 D StrictMode:    at com.android.providers.settings.SettingsProvider.getCallingApplicationInfoOrThrow(SettingsProvider.java:2182)
08-24 11:55:48.047  3333  3571 D StrictMode:    at com.android.providers.settings.SettingsProvider.enforceSettingReadable(SettingsProvider.java:2079)
08-24 11:55:48.047  3333  3571 D StrictMode:    at com.android.providers.settings.SettingsProvider.getGlobalSetting(SettingsProvider.java:1354)
08-24 11:55:48.047  3333  3571 D StrictMode:    at com.android.providers.settings.SettingsProvider.call(SettingsProvider.java:417)
08-24 11:55:48.047  3333  3571 D StrictMode:    at android.content.ContentProvider.call(ContentProvider.java:2464)
08-24 11:55:48.047  3333  3571 D StrictMode:    at android.content.ContentProvider$Transport.call(ContentProvider.java:512)
08-24 11:55:48.047  3333  3571 D StrictMode:    at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:295)
08-24 11:55:48.047  3333  3571 D StrictMode:    at android.os.Binder.execTransactInternal(Binder.java:1179)
08-24 11:55:48.047  3333  3571 D StrictMode:    at android.os.Binder.execTransact(Binder.java:1143)
```

If for you is trusted, could you try to run this command as root at boot?

`magiskpolicy --live "allow system_server system_data_root_file file { read }"`

**Reply by gamerkonks** [reply]
I've uninstalled it but still getting the issue. I'll try and get some more logs later.

While testing this morning had a weird issue.
Restarted the phone with 1.8.3 installed and left it for a while.
Got a phone call and answered. The UI completely locked up. The screen stayed on longer than timeout period and was unresponsive to any touch or button presses. The call still continued as normal.
So the UI seems to lock up on user input or after unlocking the phone.

**Reply by mywalkb** [reply]
> Please let me know if there is anymore information I can provide to help figure out what the issue is.

When this happens you could run the command as root `top -H`
You should see the top process taking up more CPU with thread name.

**Reply by mywalkb** [reply]
If you want test dex2oat, you should temporaly disable selinux with command as root `setenforce 0`

```
/ # mount | grep dex
/ # getprop | grep dalvik.vm.dex2oat-flags
[dalvik.vm.dex2oat-flags]: [--inline-max-code-units=0]
```
mount not should show nothing with dex and the prop dalvik.vm.dex2oat-flags should be set.

**Comment by mate-in-1** [comment]
Same problem with an LG V30 on Android 9 and Zygisk LSPosed 1.8.1+ / Magisk 25.1! 

Was struggling to figure out why my phone would start out ok after reboot but would get slower and slower over time to the point it was unusable. While the phone is from 2018, it actually is quite snappy without any Xposed or anything. My first instinct was to optimize all apps with Dex2oat which seems to usually help, though was only temporary in this case. 

I then uninstalled LSPosed from Magisk and my phone became snappy again. Was thinking at that point the issue was with Zygisk LSPosed and Magisk, which would mean I would have to go back to Magisk 23 for the RIRU version... 

Then I stumbled upon this discussion and installed LSPosed 1.8.0 (still Zygisk) and all is snappy again. There must be something in 1.8.1+ that is affecting some people's performance like this. 

Please let me know what logs you need and I can try and provide them. Would be unfortunate if I have to stay on 1.8.0 forever!

Thanks!
drew

**Reply by mywalkb** [reply]
Install debug version and post logcat, save zip from "Logs" in manager and post.

**Comment by mate-in-1** [comment]
Can you please provide a link to the debug version 1.8.3? The main page says to go to Actions but from there I have no clue where to go.

Then can you please provide clear step by step instructions on what you want me to do or provide a link to instructions? 

Happy to help, but please try to make it as straighforward as possible as I am effectively turning my phone into a guinea pig and leaving a working environment to go to one that I know will cause issues.

Thanks!

**Reply by mywalkb** [reply]
[This](https://github.com/LSPosed/LSPosed/actions/runs/2552167817) is the last action 1.8.3 (6577) is better to download LSPosed-v1.8.3-6577-zygisk-debug for test.
[This](https://github.com/LSPosed/LSPosed/actions/runs/2285811971) is the action with 1.8.3 (6552) last stable release.

After how long does the device become unresponsive?
Can you connect with adb before become unresponsive?

**Reply by mywalkb** [reply]
You can follow [this](https://github.com/topjohnwu/Magisk/issues/6042#issuecomment-1167132605) comment for catch log.

**Comment by frankli998** [comment]
Same here with samsung galaxy tab s7 sm-t875. I have caught log using 1.8.3 (6577)  debug version, here's log
[boot-2022-07-22-15-29-25 (2).log](https://github.com/LSPosed/LSPosed/files/9165698/boot-2022-07-22-15-29-25.2.log)
.

**Reply by mywalkb** [reply]
is better open an issue and post log file.

**Comment by yujincheng08** [comment]
`W LSPosedDex2Oat: Check mount failed for /apex/com.android.art/bin/dex2oat32`

This may be the issue

**Comment by gamerkonks** [comment]
I wonder if it could be related to this.
https://github.com/topjohnwu/Magisk/issues/3171

**Comment by m31-galaxy** [comment]
Having exactly the same issue on my Pixel 3a XL, and this is the only record of the issue I've found online. However, I can't downgrade to LSPosed 1.8.0 as it crashes on startup. If there's anything I can do to troubleshoot or provide in the way of logs, please let me know.
